lib/transaction.c

```diff
@@ -1182,16 +1182,21 @@ static int vfyCb(struct rpmsinfo_s *sinfo, void *cbdata)
     struct vfydata_s *vd = (struct vfydata_s *)cbdata;
 
     if (sinfo->type & RPMSIG_VERIFIABLE_TYPE && sinfo->rc != RPMRC_NOTFOUND) {
-	int res = (sinfo->rc != RPMRC_OK);
 	/* Take care not to override a previous failure with success */
-	if (res > vd->type[sinfo->type])
-	    vd->type[sinfo->type] = res;
+	if (sinfo->rc > vd->type[sinfo->type]) {
+	    vd->type[sinfo->type] = sinfo->rc;
+	    for (int type=0; type < 3; type++) {
+		if (type != sinfo->type && vd->type[type] >= sinfo->rc)
+		    return 1;
```

Comment:
We will still fall through (to the `switch` below and thus potentially replace the error message) if the current error in `sinfo->rc` is the same or lower than what we have stored for this type so far.